src/libraries/learningMachine/CMakeLists.txt

```diff
@@ -61,17 +60,11 @@ if(YARP_gsl_FOUND)
       src/Math.cpp 
       src/Serialization.cpp )
   
-  # declare groups of source and header files -- makes things pretty in MSVC.
-  source_group("Source Files" FILES ${LM_MACHINE_SRC} ${LM_TRANSFORMER_SRC} ${LM_SUPPORT_SRC})
-  source_group("Header Files" FILES ${LM_HEADER})
-  
-  # add YARP include directories
-  include_directories(${GSL_INCLUDE_DIRS})
-  
-  # add our include files into our compiler's search path.
-  include_directories(${PROJECT_SOURCE_DIR}/include)
-  
   add_library(${LM_LIB} ${LM_MACHINE_SRC} ${LM_TRANSFORMER_SRC} ${LM_SUPPORT_SRC} ${LM_HEADER})
+
+  target_include_directories(${LM_LIB} PUBLIC "$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include>"
+                                              "$<INSTALL_INTERFACE:$<INSTALL_PREFIX>/${CMAKE_INSTALL_INCLUDEDIR}>"
+                                              ${GSL_INCLUDE_DIRS})
```

Comment:
This will not work as intended in Windows if the binary installer is used and GSL is installed in a different location w.r.t. to the one in the machine that builds the installer, but this problem is already present so this is not a big problem.